telegram/_bot.py

```diff
@@ -2037,16 +2049,52 @@ async def send_media_group(
                 :attr:`~telegram.request.BaseRequest.DEFAULT_NONE`.
             api_kwargs (:obj:`dict`, optional): Arbitrary keyword arguments to be passed to the
                 Telegram API.
+            caption (:obj:`str`, optional): Caption that will be added to the
+                first element of :paramref:`media`, so that it will be used as caption for the
+                whole media group.
+                Defaults to :obj:`None`.
+            parse_mode (:obj:`str` | :obj:`None`, optional):
+                Parse mode for :paramref:`caption`.
+                See the constants in :class:`telegram.constants.ParseMode` for the
+                available modes.
+                Defaults to :obj:`None`.
+            caption_entities (List[:class:`telegram.MessageEntity`], optional):
+                List of special entities for :paramref:`caption`,
+                which can be specified instead of :paramref:`parse_mode`.
+                Defaults to :obj:`None`.
 
         Returns:
             List[:class:`telegram.Message`]: An array of the sent Messages.
 
         Raises:
             :class:`telegram.error.TelegramError`
         """
+        if caption and any(
+            [
+                any(item.caption for item in media),
+                any(item.caption_entities for item in media),
+                any(item.parse_mode is DEFAULT_NONE for item in media),
+            ]
+        ):
+            raise ValueError("You can only supply either group caption or media with captions.")
+
+        changed_media = None
```

Comment:
I guess one can skip this variable - just do `media = media.copy()` in the `if caption` case

Reply:
OK, I just thought that would be more readable. But yes, my way requires one more line of code and more complex "media" assignment later on.

Fixing in commit 7643dc6633fd82ab96ef9f1bd7490ddb84b7662b.

BTW, correct me if I'm wrong: as far as I understand, `media.copy()` and `media[:]` do the same thing. I'll put `media.copy()`, but just checking.

Reply:
OK, I just thought that would be more readable. But yes, my way requires one more line of code and more complex "media" assignment later on.

Fixing in commit 7643dc6633fd82ab96ef9f1bd7490ddb84b7662b.

BTW, correct me if I'm wrong: as far as I understand, `media.copy()` and `media[:]` do the same thing. I'll put `media.copy()`, but just checking.